src/Plutip/Spawn.purs

```diff
@@ -0,0 +1,73 @@
+-- | This module provides ability to spawn a program using `spawn` and wait
+-- | for some specific output that indicates that the program has started
+-- | successfully or failed, in which case an exception is thrown.
+module Plutip.Spawn
+  ( NewOutputAction(NoOp, Success, Cancel)
+  , spawnAndWaitForOutput
+  ) where
+
+import Prelude
+
+import Data.Either (Either(Left))
+import Data.Maybe (Maybe(Just, Nothing))
+import Data.Posix.Signal (Signal(SIGINT))
+import Effect (Effect)
+import Effect.Aff (Aff, Canceler(Canceler), makeAff)
+import Effect.Class (liftEffect)
+import Effect.Exception (Error, error)
+import Effect.Ref as Ref
+import Node.ChildProcess
+  ( ChildProcess
+  , SpawnOptions
+  , kill
+  , onExit
+  , spawn
+  , stdout
+  )
+import Node.Encoding as Encoding
+import Node.Stream (onDataString)
+
+-- | Provides a way to react on update of a program output.
+-- | Do nothing, indicate startup success, or thrown an exception to the Aff
+-- | action, killing the spawned program.
+data NewOutputAction = NoOp | Success | Cancel
+
+-- | `spawn`, but with ability to wait for program startup, using a callback
+-- | returning a `NewOutputAction`, or to kill the program depending on its
+-- | output.
+spawnAndWaitForOutput
+  :: String
+  -> Array String
+  -> SpawnOptions
+  -> (String -> NewOutputAction)
+  -> Aff ChildProcess
+spawnAndWaitForOutput cmd args opts filter =
+  makeAff (spawnAndWaitForOutput' cmd args opts filter)
+
+spawnAndWaitForOutput'
+  :: String
+  -> Array String
+  -> SpawnOptions
+  -> (String -> NewOutputAction)
+  -> (Either Error ChildProcess -> Effect Unit)
+  -> Effect Canceler
+spawnAndWaitForOutput' cmd args opts filter cont = do
+  child <- spawn cmd args opts
+  onExit child $ const $ cont $ Left $ error $ "Process " <> cmd <> " exited"
+  ref <- Ref.new (Just "")
+  onDataString (stdout child) Encoding.UTF8
+    \str -> do
+      output <- Ref.modify (map (_ <> str)) ref
+      case filter <$> output of
+        Nothing -> pure unit
+        Just NoOp -> pure unit
```

Comment:
could catch these as `_` although perhaps you prefer for readability which is fine.